Hash table

jeremytedwards opened the pull request.

jeremytedwards committed:
README.md updates to include added definition for BST
Added bsd.py initial framework

jeremytedwards committed:
Formatted doc strings

jeremytedwards committed:
Added test_bst stubbed out

cacizi41 committed:
Merge branch 'bst' of https://github.com/jeremytedwards/data-structures into bst

cacizi41 committed:
bst insert and test

jeremytedwards committed:
Added test_bst stubbed out

jeremytedwards committed:
working though the insert left and right

cacizi41 committed:
some update, not working yet

cacizi41 committed:
insert and test working

cacizi41 committed:
contains and test working

cacizi41 committed:
size and depth working

cacizi41 committed:
balance and test working

jeremytedwards committed:
Updated comments and spacing.

jeremytedwards committed:
Updated insert test to test for duplicates.

jeremytedwards committed:
Added InOrder
Added PreOrder
Added PostOrder

jeremytedwards committed:
Working on tests

cacizi41 committed:
traversal and test pass

jeremytedwards committed:
Working on tests

cacizi41 committed:
traversal and tests working

cacizi41 committed:
clean up codeg

jeremytedwards committed:
Merge branch 'bst-traversal' of https://github.com/jeremytedwards/data-structures into bst

# Conflicts:
#	tests/test_bst.py

jeremytedwards committed:
Added _balance function
Added _gen_tree_from_ord_list
Starting to stub out the delete function
Added tests for above

cacizi41 committed:
deletion working except root delete case

jeremytedwards committed:
Added _reset_by_value function to the Node
Writing tests to validate tree.delete works for a tree of a single node

jeremytedwards committed:
Cleaned up the docstrings

jeremytedwards committed:
Added Travis.yml file

jeremytedwards committed:
Updated Travis.yml file to run only 3.5

jeremytedwards committed:
Updated Travis.yml file to run both py27,py35

jeremytedwards committed:
Updated Travis.yml file to run both py35

cacizi41 committed:
work on self-balancing

jeremytedwards committed:
Updates to README.md
Revised insert to insert with recursion

jeremytedwards committed:
Revisions to node_balance

jeremytedwards committed:
Revisions to node_balance... again

cacizi41 committed:
self balancinggggg

jeremytedwards committed:
Stubbed out the assigment for test and hash_table

jeremytedwards committed:
Added the _hash helper method to take a string or int

jeremytedwards committed:
Wrote some tests for _hash
Wrote some tests for __init__

jeremytedwards committed:
Writing the set function

jeremytedwards committed:
Revisions to the set method

jeremytedwards committed:
Updated the get method

cacizi41 committed:
more tests on get

cacizi41 committed:
set func if statement not working

cacizi41 committed:
set func modified

jeremytedwards committed:
Working though BST insertion balance
Paramaterized the Set Test for Hash Table

jeremytedwards committed:
Added test for TypeError on set()
Wrote type error into set()

jeremytedwards committed:
Revision to type error into set()

jeremytedwards committed:
Tested words from /usr/share/dict/words

jeremytedwards committed:
Revisions to README

nhuntwalker commented on src/data_structures/hash_table.py:
```diff
@@ -0,0 +1,38 @@
+        hashes the key provided(note that this is an internal api)
+        this method prefers a string or integer.
+        """
+        return sum([ord(c) for c in str(key)]) % self._size
```
You're using the additive hash function. You should note that in your doc strings somewhere (or your README)

nhuntwalker commented on tests/test_hash_table.py:
```diff
@@ -0,0 +1,108 @@
+    from data_structures.hash_table import HashTable
+    test_hash_table = HashTable(hash_size=0)
+    assert len(test_hash_table._storage) == 0
+
```
Test that if you try to use a key that isn't currently in your hash table, you return None

jeremytedwards committed:
Revised tests for work in progress, added TODOs

jeremytedwards committed:
Revised tests for work in progress, added TODOs

jeremytedwards committed:
Added to doc string and README the definition of the hashing type used.

jeremytedwards committed:
Test added for case: None

jeremytedwards committed:
Repaired tests.

Merge branch 'bst' of https://github.com/jeremytedwards/data-structures into bst

# Conflicts:
#	tests/test_bst.py

jeremytedwards committed:
Increased BST test coverage

Merge branch 'bst' of https://github.com/jeremytedwards/data-structures into bst

# Conflicts:
#	tests/test_bst.py

jeremytedwards committed:
revisions to tests to get them passing.
Added "words" file for hash_table testing

Merge branch 'bst' of https://github.com/jeremytedwards/data-structures into hash-table

# Conflicts:
#	src/data_structures/bst.py
#	tests/test_bst.py

jeremytedwards committed:
Update to tests for "words" file
